### PR TITLE
feat(share/shrex/nd): add swamp test for shrex nd

### PR DESCRIPTION
**Pull request by walldiss**
## Overview
Feel free to suggest more test cases if you have any ideas

Closes https://github.com/celestiaorg/celestia-node/issues/2119

### CODE REVIEW

File: nodebuilder/tests/nd_test.go
Comment (walldiss) [reply nodebuilder/tests/nd_test.go:162]: Not sure, but we need him.

File: share/p2p/shrexnd/server.go
Comment (walldiss) [reply share/p2p/shrexnd/server.go:69]: Nice find!

**Comment by Bidon15** [comment]
Fantastic PR @walldiss 🚀 

I believe you can close this issue https://github.com/celestiaorg/celestia-node/issues/2119 with this PR

**Review by renaynay — COMMENTED** [review]
@walldiss let's convert this to draft / just wait until cleanussy is merged so tht you can follow the new cleanussy pattern and rely on some cleaner endpoints i have

File: nodebuilder/share/config.go
Comment (walldiss) [reply nodebuilder/share/config.go:35]: Good suggestion. Agree 100%

File: nodebuilder/share/constructors.go
Comment (walldiss) [reply nodebuilder/share/constructors.go:72]: Order and structure is important here, so I would rather not. With this approach we define topology of getters in single place and it is much easier to understand like that. For example we wrap ipld and shrex getter with teegetter, but don't wrap the store getter.

**Comment by renaynay** [comment]
During a random run of `TestShrexNDFromLightsWithBadFulls`, got this fatal:

```
2023-07-31T14:43:18.569+0200	DEBUG	share/getters	getters/cascade.go:115	cascade: launching getter #0
2023-07-31T14:43:18.570+0200	ERROR	ipld	ipld/namespace_data.go:247	could not retrieve IPLD node	{"namespace": "00000000000000000000000000000000000000186d38b0311dc2a45a12", "pos": 0, "err": "nmt node not found"}
2023-07-31T14:43:18.570+0200	ERROR	ipld	ipld/namespace_data.go:247	could not retrieve IPLD node	{"namespace": "00000000000000000000000000000000000000186d38b0311dc2a45a12", "pos": 0, "err": "nmt node not found"}
2023-07-31T14:43:18.570+0200	ERROR	ipld	ipld/namespace_data.go:247	could not retrieve IPLD node	{"namespace": "00000000000000000000000000000000000000186d38b0311dc2a45a12", "pos": 1, "err": "nmt node not found"}
2023-07-31T14:43:18.570+0200	ERROR	ipld	ipld/namespace_data.go:247	could not retrieve IPLD node	{"namespace": "00000000000000000000000000000000000000186d38b0311dc2a45a12", "pos": 0, "err": "nmt node not found"}
2023-07-31T14:43:18.570+0200	FATAL	ipld	ipld/namespace_data.go:305	target namespace outside of boundaries of links at depth: 3
```

**Review by renaynay — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]

**Pull request merged**
